### PR TITLE
Improve auth and allow PATCHing users

### PR DESCRIPTION
**Pull request by philectron**
- Remove user's role from JWT payload; instead, fetch user directly from DB and check for latest role
- Add `PATCH /user/:userId` route
- Add `userId`'s constraints to `userSchema` and apply additional checks to `userId`
- Update DB init script with 11-digit `userId` for dummy users

### CODE REVIEW

**Comment by philectron** [comment]
I _think_ I completely replaced checking the role in the JWT with directly fetching the user from the DB on all instances of `req.auth` (by doing a global search). But still, keep an eye out for `req.auth.userRole` or `auth.userRole` in case I haven't got rid of all of them.

File: api/plan.js [hunk 1]
```diff
@@ -22,6 +22,7 @@ const {
   deletePlan,
```
Comment (silverware13) [inline api/plan.js]: This version of api/plan.js is quite a bit behind master.
The function you are modifying doesn't exist anymore, and there are no references to req.auth.userRole in the current version of api/plan.js so there isn't a need to require("../models/user"); in this file either.

I would suggest simply using the current version of api/plan.js from master.

**Review by silverware13 — CHANGES_REQUESTED** [review]
Everything looks good to me with the exception of the api/plan.js.

The version you are editing is out of date and the version on master doesn't use req.auth.userRole or the API endpoint being edited.

I would request simply using the updated version of api/plan.js from master.

File: api/plan.js (hunk 1 above)
Comment (philectron) [reply api/plan.js]: Done.

**Review by silverware13 — APPROVED** [review]
Looks good.

**Pull request merged**